cmd/kubeadm/app/cmd/version.go

```diff
@@ -35,10 +44,39 @@ func NewCmdVersion(out io.Writer) *cobra.Command {
 			kubeadmutil.CheckErr(err)
 		},
 	}
+	cmd.Flags().BoolP("short", "", false, "only print the version (tag|number)")
+	cmd.Flags().StringP("output", "o", "", "output format, options available are yaml and json")
 	return cmd
 }
 
 func RunVersion(out io.Writer, cmd *cobra.Command) error {
-	fmt.Fprintf(out, "kubeadm version: %#v\n", version.Get())
+
+	v := Version{
+		clientVersion: &version.Get(),
+	}
+
+	switch of := cmdutil.GetFlagString(cmd, "output"); of {
+	case "":
+		if cmdutil.GetFlagBool(cmd, "short") {
+			fmt.Fprintf(out, "kubeadm version: %s\n", v.clientVersion.GitVersion)
```

Comment:
if the user really wants the short version, we should probably drop `kubeadm version` here